### ISSUE TITLE
Training my own dataset, metrics like AP are too low at train all.

### ISSUE THREAD

**Issue by huanghuang113** [VinAIResearch/ISBNet#32]
Hi, first of all thank you very much for your great contribution in the direction of point cloud segmentation. I processed my own data files according to stpls3d, but my data files are smaller, they are plant point cloud data, so I processed them with both train and val_250m `torch.save((coords, colors, sem_labels, instance_labels) , pth_file)` saves them as .pth files without chunking them. When training my backbone, I only did 12 epochs and here are the results of my backbone run.
![image](https://github.com/VinAIResearch/ISBNet/assets/119155412/20a027f7-ea72-4c79-b795-422c151d75f0)

But when I do the TRAIN ALL, the result is not very good, the evaluation at the end of the first epoch is too bad as below:
![image](https://github.com/VinAIResearch/ISBNet/assets/119155412/3eb3e6f7-98ca-4273-aacd-e99f0519dcd5)

And after 12 epochs, the results are not very good, as shown below:
![image](https://github.com/VinAIResearch/ISBNet/assets/119155412/df2e0aab-4d93-4489-8a97-b4ede2bce67e)

Here are my config file settings:
model:
  channels: 16
  num_blocks: 7
  semantic_classes: 2
  instance_classes: 2
  sem2ins_classes: []
  semantic_only: False
  semantic_weight: [1.0, 1.0]
  with_coords: False
  ignore_label: -100
  voxel_scale: 50
  use_spp_pool: False
  filter_bg_thresh: 0.1
  iterative_sampling: False
  mask_dim_out: 16
  instance_head_cfg:
    dec_dim: 64
    n_sample_pa1: 2048
    n_queries: 256
    radius_scale: 10
    radius: 0.04
    neighbor: 16
  test_cfg:
    x4_split: False
    logit_thresh: 0.0
    score_thresh: 0.2
    npoint_thresh: 10
    type_nms: 'matrix'
    topk: 100

  fixed_modules: ['input_conv', 'unet', 'output_layer', 'semantic_linear', 'offset_linear', 'offset_vertices_linear', 'box_conf_linear']

data:
  train:
    type: 'plant'
    data_root: 'dataset/plant'
    prefix: 'train1'
    suffix: '.pth'
    training: True
    repeat: 3
    voxel_cfg:
      scale: 50
      spatial_shape: [128, 512]
      max_npoint: 250000
      min_npoint: 5000
  test:
    type: 'plant'
    data_root: 'dataset/plant'
    prefix: 'val_250m1'
    suffix: '.pth'
    training: False
    voxel_cfg:
      scale: 50
      spatial_shape: [128, 512]
      max_npoint: 250000
      min_npoint: 5000

dataloader:
  train:
    batch_size: 8
    num_workers: 8
  test:
    batch_size: 1
    num_workers: 1

optimizer:
  type: 'AdamW'
  lr: 0.001
  weight_decay: 0.0001

save_cfg:
  semantic: False
  offset: False
  instance: True
  offset_vertices: False
  nmc_clusters: False
  object_conditions: False

fp16: True
epochs: 12
step_epoch: 4
save_freq: 4
pretrain: 'work_dirs/plant/isbnet_backbone_plant/exp3/latest.pth'
work_dir: ''

Can you give me some advice please, thank you very much!

**Comment by Endvour** [comment]
have you solved the problem?

**Comment by huanghuang113** [comment]
> I also have this question.

Train for a few more epochs and you'll be fine, the accuracy will get higher at about the 20th-30th epoch.

**Comment by huanghuang113** [comment]
Maybe you didn't set the (def get_instances) method correctly.

**Comment by huanghuang113** [comment]
I was using ISBNet by modifying it here (isbnet/model/isbnet.py/get_instance)
